### PR TITLE
Build dockerfile at CI trigger

### PR DESCRIPTION
**Pull request by sukyoungjeong-furiosa**
### Motivation

Support development for both major and minor releases (which requires distinct toolchains) by enabling toolchain changes during CI without modifying the infrastructure.

### Changes

`furiosa-models-build`
= `clone` - `lint` - **`build-and-push`** - `unittests` & `test-examples`

- Build and push a docker image before tests. The image contains required toolchains and `furiosa-models` package (of the commit) installed. Reuse the image for subsequent steps and regression-tests.
- Make use of `toochain` target of the Makefile at CI.
- Bring a minimal directory structure to `.ci` folder.

### Notes

- Fixed libhal version as 0.9.0-2. (0.10.0-2 seems to have driver issue with warboy a0, causing `enf configuration is not compatible with NPU` error.)
  - This may be fixed if we start using warboy b0 as our default target.

### TODO

- Pruning images periodically.
- Pruning `pipelinerun`s, `taskrun`s periodically.
- Fast up the docker build step.
- Force the regression test to retry pulling image when it fails to.

### CODE REVIEW

**Comment by furiosa-infra** [comment]
```

--------------------------------------------------------------------------------------------------------- benchmark: 8 tests ---------------------------------------------------------------------------------------------------------
Name (time in ms)                                                  Min                   Max                Mean             StdDev              Median                IQR            Outliers       OPS            Rounds  Iterations
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_mlcommons_resnet50_accuracy                                3.1905 (1.0)         81.7717 (3.95)       7.5370 (1.11)      1.6356 (1.81)       7.2881 (1.11)      1.2186 (1.86)    5947;3037  132.6783 (0.90)      50000           1
test_mlcommons_resnet50_with_native_pp_accuracy                 3.3067 (1.04)        72.1861 (3.49)       6.7812 (1.0)       1.4321 (1.58)       6.5850 (1.0)       0.6566 (1.0)     3979;4671  147.4662 (1.0)       50000           1
test_mlcommons_ssd_mobilenet_with_native_rust_pp_accuracy       5.1620 (1.62)        20.6984 (1.0)        7.3009 (1.08)      0.9040 (1.0)        7.1154 (1.08)      0.7851 (1.20)      734;243  136.9693 (0.93)       5000           1
test_yolov5m_accuracy                                           8.5265 (2.67)       211.1230 (10.20)     12.7634 (1.88)      3.9785 (4.40)      12.3132 (1.87)      1.7127 (2.61)      220;384   78.3491 (0.53)      10000           1
test_mlcommons_ssd_mobilenet_accuracy                           8.6948 (2.73)        40.7444 (1.97)      13.1676 (1.94)      2.1446 (2.37)      12.6437 (1.92)      1.7311 (2.64)      770;368   75.9441 (0.51)       5000           1
test_yolov5l_accuracy                                          11.3381 (3.55)       205.1856 (9.91)      17.0487 (2.51)      4.9661 (5.49)      16.4504 (2.50)      3.0608 (4.66)      538;386   58.6556 (0.40)      10000           1
test_mlcommons_ssd_resnet34_with_native_rust_pp_accuracy      103.0201 (32.29)      496.4072 (23.98)    130.0155 (19.17)    12.3019 (13.61)    132.3995 (20.11)    11.0820 (16.88)     1015;88    7.6914 (0.05)       5000           1
test_mlcommons_ssd_resnet34_accuracy                          144.0233 (45.14)    1,047.3833 (50.60)    190.5793 (28.10)    32.7237 (36.20)    185.4087 (28.16)    22.5939 (34.41)     397;232    5.2472 (0.04)       5000           1
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
```

File: .ci/furiosa-models-ci/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+metadata:
+  name: lint
+  namespace: ci-furiosa-models
+spec:
+  workspaces:
+    - name: source
+  params:
+    - name: image
+      description: The container image to use black in
+      default: asia-northeast3-docker.pkg.dev/next-gen-infra/furiosa-ai/furiosa-models:v0.8.0
```
Comment (furiosamg) [inline .ci/furiosa-models-ci/lint.yaml:12]: Just being curious, we don't need this specific tag right?

**Review by furiosamg — APPROVED** [review]
Looks good to me.

File: .ci/furiosa-models-ci/lint.yaml (hunk 1 above)
Comment (sukyoungjeong-furiosa) [reply .ci/furiosa-models-ci/lint.yaml:12]: I'll add another tag `lint` for that image and make use of it.

**Pull request merged**